angular/src/app/shared/data-table/data-table.component.ts

```diff
@@ -30,15 +31,27 @@ export class DataTableComponent {
   rowSelection: TableRowSelection = TableRowSelection.single;
   @Input()
   pagination = true;
+  @Input()
+  deselectRowsEvents: Subject<void> = new Subject();
```

Comment:
I usually try to **avoid passing Observables down to child components**. You can find more about it when you start searching for this topic. https://angular.io/guide/component-interaction

The way I would do it is either use passing simple `boolean` and reacting on `ngOnChanges` (which usually is enough), or keeping the way you do it, but moving the logic to service (e.g. `data-table-events.service.ts`) and sharing the service between the parent (who is deciding when to deselect) and child (`data-table` who is executing orders which are "stored" in service). This way you can pass to `data-table` any event and it is easily extendable, it also solves the problem where you are creating artifical input for `beacon-search-result-table` just to pass it to `data-table`.

Reply:
Thanks for the feedback. I'm still not super familiar with angular patterns yet, so feedback like this helps a lot. That all makes sense, I'll clean it up.